src/RelayServer.ts

```diff
@@ -377,40 +353,50 @@ export class RelayServer extends EventEmitter {
                 )
             );
         }
-        log.debug(
-            'RequestFees - allowForSponsoredTx ',
-            this.config.allowForSponsoredTx
-        );
-        if (!this.config.allowForSponsoredTx) {
+
+        const { workerFeePercentage } = this.config;
+        log.debug(`RelayServer - workerFeePercentage: ${workerFeePercentage}`);
+
+        const sponsoredTxFee: BigNumber = normaliseFraction({ fraction: workerFeePercentage }).absoluteValue();
+        if (sponsoredTxFee.isGreaterThan(0)) {
```

Comment:
Although it makes sense. What if still want to perform the check of the received token against the estimation performed server side without adding any percentage? I think that practically this would be achieved by setting a very low `workerFeePercentage`, but maybe we could make it clearer. WDYT?

Reply:
changed